name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/util/ClientFilterMenu.java

```diff
@@ -276,12 +284,10 @@ public Image getImage(Object element)
 
     private Optional<Item> buildItem(String uuids)
     {
-        if (uuids == null || uuids.isEmpty())
+        if (uuids == null)
             return Optional.empty();
 
         String[] ids = uuids.split(","); //$NON-NLS-1$
-        if (ids.length == 0)
-            return Optional.empty();
```

Comment:
What is the reason to delete those checks?

Reply:
With these checks the filter is implicit deleted when the user removes all child elements from main filter.
When deleting all child elements from main filter, saving file and restart PP the filter is not available in filter list anymore. In my opinion this behaviour will confuse the user, because they have no information (message box etc) why the main filter is missing.

The other question is: how often will this case (user deletes all child elements) happen? But my experience is: when there is even a small chance that something can happen, it will happen (sooner or later).